homeassistant/components/script.py

```diff
@@ -42,6 +44,62 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+_ALIAS_VALIDATOR = vol.Schema(cv.string)
+
+
+def _alias_stripper(validator):
+    """Strip alias from object for validation."""
+    def validate(value):
+        """Validate without alias value."""
+        value = value.copy()
+        alias = value.pop(CONF_ALIAS, None)
+
+        if alias is not None:
+            alias = _ALIAS_VALIDATOR(alias)
+
+        value = validator(value)
+
+        if alias is not None:
+            value[CONF_ALIAS] = alias
+
+        return value
+
+    return validate
+
+
+_DELAY_SCHEMA = {
+    vol.Required(CONF_DELAY): vol.All({
+        CONF_ALIAS: str,
+        'days': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'seconds': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'microseconds': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'milliseconds': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'minutes': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'hours': vol.All(vol.Coerce(int), vol.Range(min=0)),
+        'weeks': vol.All(vol.Coerce(int), vol.Range(min=0)),
+    }, cv.has_at_least_one_key([
+        'days', 'seconds', 'microseconds', 'milliseconds', 'minutes', 'hours',
+        'weeks']))
+}
+
+_EVENT_SCHEMA = cv.EVENT_SCHEMA.extend({
+    CONF_ALIAS: cv.string,
+})
+
+_SCRIPT_ENTRY_SCHEMA = vol.Schema({
+    CONF_ALIAS: str,
+    vol.Required(CONF_SEQUENCE): vol.All(vol.Length(min=1), [vol.Any(
```

Comment:
Length will validate any type with a length such as a string, so this might need an additional `list` at the beginning.
Also, AFAIK, the list doesn't validate that it is a list and would be equivalent to vol.Any, so this reads to me like `vol.Any(vol.Any(...))` but I might just be reaching my knowledge of the subtleties of voluptuous here.